cli/run_test.go

```diff
@@ -0,0 +1,176 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var dispatchBinary = filepath.Join("../build", runtime.GOOS, runtime.GOARCH, "dispatch")
+
+func TestRunCommand(t *testing.T) {
+	t.Run("Run with non-existent env file", func(t *testing.T) {
+		t.Parallel()
+
+		buff, msg, err := execRunCommand(&[]string{}, "run", "--env-file", "non-existent.env", "--", "echo", "hello")
```

Comment:
This test could be improved by using a temporary directory to store the env file; you can create one with [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir).

Reply:
This one doesn't create a file, it tests that we throw an error if file doesn't exist. I use `t.TempDir()` in the tests where the file is actually being created

Reply:
We should be able to control where the files get created right?

You can compose the full path with `filepath.Join(tmpdir, filename)` for example.

Reply:
This test makes sure we get correct error when the value of `--env-file` is not a valid path. We don't need to create any temporary files or directories in this test.

Reply:
Ah! My bad, I misunderstood this test :)